src/notation/qml/MuseScore/NotationScene/internal/DynamicPopup.qml

```diff
@@ -0,0 +1,213 @@
+import QtQuick 2.15
+import QtQuick.Layouts 1.15
+
+import Muse.Ui 1.0
+import Muse.UiComponents 1.0
+import MuseScore.NotationScene 1.0
+
+StyledPopupView {
+    id: root
+
+    property NavigationSection notationViewNavigationSection: null
+    property int navigationOrderStart: 0
+    property int navigationOrderEnd: dynamicsNavPanel.order
+
+    property int buttonHeight: 30
+
+    property int currentPage: 0
+    property bool isHairpinPage: currentPage == 6
+
+    property var widthOffsetList: [
```

Comment:
I think it is not ideal that this information is so "far away" from the dynamics information itself. This will require carefully keeping `DYN_NAMES` in C++ in sync with this `widthOffsetList` here in QML.

Of course, it isn't really possible to move everything to QML, because of the necessary IEngravingFont calculations that can't be done in QML. So you will need to go the other way around: move this data to C++. 

I think it would be good to put this in the `DYN_NAMES` list (and perhaps then update the name of that list, since it's not only the names anymore). Instead of only a QString, you could store a `struct` in that list, which contains the dynamic name, the width, and the offset.

For QML, you could then expose these values as a QVariantMap, for example like this:
```
{
    { "symbol": /* what you get from IEngravingFont's toString */ },
    { "width": … },
    { "offset": … }
}